services/rfq/relayer/pricer/fee_pricer_test.go

```diff
@@ -207,21 +211,55 @@ func (s *PricerSuite) TestGetTotalFee() {
 	headerOrigin := big.NewInt(100_000_000_000)      // 100 gwei
 	headerDestination := big.NewInt(500_000_000_000) // 500 gwei
-	clientOrigin.On(testsuite.GetFunctionName(clientOrigin.SuggestGasPrice), mock.Anything).Once().Return(headerOrigin, nil)
-	clientDestination.On(testsuite.GetFunctionName(clientDestination.SuggestGasPrice), mock.Anything).Once().Return(headerDestination, nil)
-	clientFetcher.On(testsuite.GetFunctionName(clientFetcher.GetClient), mock.Anything, big.NewInt(int64(s.origin))).Once().Return(clientOrigin, nil)
-	clientFetcher.On(testsuite.GetFunctionName(clientFetcher.GetClient), mock.Anything, big.NewInt(int64(s.destination))).Once().Return(clientDestination, nil)
+	clientOrigin.On(testsuite.GetFunctionName(clientOrigin.SuggestGasPrice), mock.Anything).Return(headerOrigin, nil)
+	clientDestination.On(testsuite.GetFunctionName(clientDestination.SuggestGasPrice), mock.Anything).Return(headerDestination, nil)
+	clientFetcher.On(testsuite.GetFunctionName(clientFetcher.GetClient), mock.Anything, big.NewInt(int64(s.origin))).Return(clientOrigin, nil)
+	clientFetcher.On(testsuite.GetFunctionName(clientFetcher.GetClient), mock.Anything, big.NewInt(int64(s.destination))).Return(clientDestination, nil)
 	priceFetcher := getPriceFetcher(nil)
-	feePricer := pricer.NewFeePricer(s.config, clientFetcher, priceFetcher, metrics.NewNullHandler())
+	feePricer := pricer.NewFeePricer(s.config, clientFetcher, priceFetcher, metrics.NewNullHandler(), relayerAddress)
 	go func() { feePricer.Start(s.GetTestContext()) }()
 
 	// Calculate the total fee.
-	fee, err := feePricer.GetTotalFee(s.GetTestContext(), s.origin, s.destination, "USDC", true)
+	fee, err := feePricer.GetTotalFee(s.GetTestContext(), s.origin, s.destination, "USDC", true, nil)
 	s.NoError(err)
 
 	// The expected fee should be the sum of the Origin and Destination fees, i.e. 100_250_000.
 	expectedFee := big.NewInt(100_250_000) // 100.25 usd
 	s.Equal(expectedFee, fee)
+
+	// Calculate the total fee with v2 call value.
+	quoteRequest := &reldb.QuoteRequest{
+		RawRequest: []byte{},
+		Transaction: fastbridgev2.IFastBridgeV2BridgeTransactionV2{
+			ZapNative: big.NewInt(1 * 1e18),
+		},
+	}
+	fee, err = feePricer.GetTotalFee(s.GetTestContext(), s.origin, s.destination, "USDC", true, quoteRequest)
+	s.NoError(err)
+
+	// The expected fee should be the sum of the Origin and Destination fees, i.e. 100_750_000,
+	// plus the call value of 1 MATIC * 0.5 USD.
+	expectedFee = big.NewInt(100_750_000) // 100.75 usd
+	s.Equal(expectedFee, fee)
+
+	// Calculate the total fee with v2 call value and call params.
+	quoteRequest = &reldb.QuoteRequest{
+		RawRequest: []byte{},
+		Transaction: fastbridgev2.IFastBridgeV2BridgeTransactionV2{
+			ZapNative:  big.NewInt(1 * 1e18),
+			ZapData:    []byte{1},
+			DestAmount: big.NewInt(1 * 1e18),
+		},
+	}
+	clientDestination.On(testsuite.GetFunctionName(clientDestination.EstimateGas), mock.Anything, mock.Anything).Once().Return(uint64(1_000_000), nil)
```

Comment:
Likewise, using empty `RawRequest` during the on-chain call will result in a revert, so we should be using a more accurate mocking strategy (if any). My bet would be once the pricer correctly constructs the `RawRequest`, the gas estimation will just work without any mocking required.

Reply:
This is just a unit test with no anvil backend for on-chain interactions- IMO we are fine to mock here since the goal of these tests is just to make sure that the calls are made properly- anything relying on the actual chain interactions should be an e2e test